### PR TITLE
Group chat reaction bug fix

### PR DESCRIPTION
**Pull request by veken1199**
1- Added the address of the sender to the data message in order to prevent errors when the app tries to parse server address as regular address
2- added some logic in media messages sender pipeline since group chat reactions and messages are sent as media messages instead of regular text messages

### CODE REVIEW

**Review by Radu-Raicea — DISMISSED** [review]

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 1]
```diff
@@ -788,9 +788,20 @@ public void onEmojiSelected(String emoji) throws InvalidMessageException {
...
 
+        String myAddress = "";
+        IdentityDatabase identityDatabase = DatabaseFactory.getIdentityDatabase(getApplicationContext());
+        try {
+          myAddress = identityDatabase.getMyIdentity().getAddress().serialize();
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
+
         String body = "{\"type\": \"reaction\", \"hash\": \"" + messageRecord.getHash() + "\", \"emoji\": \"" +
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/ConversationActivity.java]: Think this line could get much clearer if we were to use some JSON serialization from a map

File: src/org/thoughtcrime/securesms/jobs/PushTextSendJob.java [hunk 2]
```diff
@@ -57,7 +58,7 @@ public void onPushSend(MasterSecret masterSecret) throws NoSuchMessageException,
       deliver(record);
 
       String messageBody = record.getDisplayBody().toString();
-      if (messageBody.length() >= 19 && messageBody.substring(0, 19).equals("{\"type\": \"reaction\"")) {
+      if (!Stereotype.fromBody(messageBody).equals(Stereotype.UNKNOWN)) {
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/jobs/PushTextSendJob.java:61]: Damn, `Stereotype` is such a convenient class wonder who the all-star pair programmers who wrote it are

File: src/org/thoughtcrime/securesms/ConversationActivity.java (hunk 1 above)
Comment (Radu-Raicea) [reply src/org/thoughtcrime/securesms/ConversationActivity.java]: I agree, I'll make this change

File: src/org/thoughtcrime/securesms/jobs/PushTextSendJob.java (hunk 2 above)
Comment (veken1199) [reply src/org/thoughtcrime/securesms/jobs/PushTextSendJob.java:61]: hahahah

File: src/org/thoughtcrime/securesms/ConversationActivity.java (hunk 1 above)
Comment (veken1199) [reply src/org/thoughtcrime/securesms/ConversationActivity.java]: yess we could have a builder class or something that sets all the attrs

File: src/org/thoughtcrime/securesms/jobs/PushTextSendJob.java (hunk 2 above)
Comment (Radu-Raicea) [reply src/org/thoughtcrime/securesms/jobs/PushTextSendJob.java:61]: Yeah sick job guys!

File: src/org/thoughtcrime/securesms/ConversationActivity.java (hunk 1 above)
Comment (Radu-Raicea) [reply src/org/thoughtcrime/securesms/ConversationActivity.java]: Pushed the change

**Review by Radu-Raicea — APPROVED** [review]

**Comment by veken1199** [comment]
Close #148

File: src/org/thoughtcrime/securesms/database/MmsDatabase.java [hunk 3]
```diff
@@ -930,8 +931,10 @@ private long insertMediaMessage(@NonNull MasterSecretUnion masterSecret,
         insertListener.onComplete();
       }
 
-      notifyConversationListeners(contentValues.getAsLong(THREAD_ID));
-      DatabaseFactory.getThreadDatabase(context).update(contentValues.getAsLong(THREAD_ID), true);
+      if(Stereotype.fromBody(body).equals(Stereotype.UNKNOWN)) {
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/database/MmsDatabase.java]: @g-harel @Radu-Raicea this should be  "!" NOT how can we fix it  ???

File: src/org/thoughtcrime/securesms/jobs/PushGroupSendJob.java [hunk 4]
```diff
@@ -87,14 +88,18 @@ public void onPushSend(MasterSecret masterSecret)
     try {
       deliver(masterSecret, message, filterAddress == null ? null : Address.fromSerialized(filterAddress));
 
-      database.markAsSent(messageId, true);
-      markAttachmentsUploaded(messageId, message.getAttachments());
+      if (Stereotype.fromBody(message.getBody()).equals(Stereotype.UNKNOWN)) {
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/jobs/PushGroupSendJob.java:91]: @g-harel  @Radu-Raicea this should be "!" NOT how can we fix it ???

**Pull request merged**